{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "811bf9d9",
   "metadata": {},
   "source": [
    "# Import, options"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "c7a0dcfb",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy, matplotlib, pandas, scanpy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "01ebcba6",
   "metadata": {},
   "outputs": [],
   "source": [
    "matplotlib.rcParams.update({'font.size':20, 'xtick.labelsize':20, 'ytick.labelsize':20, \n",
    "                            'axes.grid' : True, 'grid.alpha': 0.5, 'grid.linestyle' : ':',\n",
    "                            'figure.figsize':(8, 5), 'svg.fonttype' : 'none'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "9b6430a8",
   "metadata": {},
   "outputs": [],
   "source": [
    "file_directory = \"/Users/kja11/OneDrive - Menntaský/PostDoc_Hypothermia/in_silico/Python/\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a9797af2",
   "metadata": {},
   "source": [
    "# Data Downloads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "e7d0584e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(16062, 15) \n",
      " Index(['Row.names', 'baseMean', 'log2FoldChange', 'lfcSE', 'stat', 'pvalue',\n",
      "       'padj', 'gene', '37_degrees', '37_degrees.1', '37_degrees.2',\n",
      "       '32_degrees', '32_degrees.1', '32_degrees.2', '32_degrees.3'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padj</th>\n",
       "      <th>lfcSE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Cirbp</td>\n",
       "      <td>2.021396</td>\n",
       "      <td>4.305259e-145</td>\n",
       "      <td>6.939647e-141</td>\n",
       "      <td>0.078809</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Xbp1</td>\n",
       "      <td>-1.884029</td>\n",
       "      <td>4.131947e-140</td>\n",
       "      <td>3.330143e-136</td>\n",
       "      <td>0.074767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Hspa8</td>\n",
       "      <td>-1.231949</td>\n",
       "      <td>4.968957e-92</td>\n",
       "      <td>2.669821e-88</td>\n",
       "      <td>0.060548</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Dnaja3</td>\n",
       "      <td>-1.053931</td>\n",
       "      <td>9.850430e-75</td>\n",
       "      <td>3.969477e-71</td>\n",
       "      <td>0.057622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Tra2a</td>\n",
       "      <td>-2.158750</td>\n",
       "      <td>2.895976e-68</td>\n",
       "      <td>9.336048e-65</td>\n",
       "      <td>0.123641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16057</th>\n",
       "      <td>Pex2</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>9.991703e-01</td>\n",
       "      <td>9.994484e-01</td>\n",
       "      <td>0.094263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16058</th>\n",
       "      <td>Ank3</td>\n",
       "      <td>-0.000171</td>\n",
       "      <td>9.993065e-01</td>\n",
       "      <td>9.994925e-01</td>\n",
       "      <td>0.196786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16059</th>\n",
       "      <td>Hmgcl</td>\n",
       "      <td>0.000057</td>\n",
       "      <td>9.994949e-01</td>\n",
       "      <td>9.995569e-01</td>\n",
       "      <td>0.090005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16060</th>\n",
       "      <td>Arl6ip5</td>\n",
       "      <td>0.000070</td>\n",
       "      <td>9.994792e-01</td>\n",
       "      <td>9.995569e-01</td>\n",
       "      <td>0.107616</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16061</th>\n",
       "      <td>Wdr7</td>\n",
       "      <td>-0.000116</td>\n",
       "      <td>9.999382e-01</td>\n",
       "      <td>9.999382e-01</td>\n",
       "      <td>1.494583</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16062 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          gene  log2FoldChange         pvalue           padj     lfcSE\n",
       "0        Cirbp        2.021396  4.305259e-145  6.939647e-141  0.078809\n",
       "1         Xbp1       -1.884029  4.131947e-140  3.330143e-136  0.074767\n",
       "2        Hspa8       -1.231949   4.968957e-92   2.669821e-88  0.060548\n",
       "3       Dnaja3       -1.053931   9.850430e-75   3.969477e-71  0.057622\n",
       "4        Tra2a       -2.158750   2.895976e-68   9.336048e-65  0.123641\n",
       "...        ...             ...            ...            ...       ...\n",
       "16057     Pex2        0.000098   9.991703e-01   9.994484e-01  0.094263\n",
       "16058     Ank3       -0.000171   9.993065e-01   9.994925e-01  0.196786\n",
       "16059    Hmgcl        0.000057   9.994949e-01   9.995569e-01  0.090005\n",
       "16060  Arl6ip5        0.000070   9.994792e-01   9.995569e-01  0.107616\n",
       "16061     Wdr7       -0.000116   9.999382e-01   9.999382e-01  1.494583\n",
       "\n",
       "[16062 rows x 5 columns]"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#data mouse_NPC\n",
    "mouse_NPC_df = pandas.read_csv(file_directory+'1) input/mNPC_all_detected_genes_DEseq.csv',\n",
    "                               sep = ',')\n",
    "print(mouse_NPC_df.shape, '\\n', mouse_NPC_df.columns)\n",
    "\n",
    "mouse_NPC_df = mouse_NPC_df[['gene', 'log2FoldChange', 'pvalue', 'padj', 'lfcSE']]\n",
    "mouse_NPC_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "15c8f337",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>Score</th>\n",
       "      <th>FDR</th>\n",
       "      <th>GeCKO_number</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>STXBP1</td>\n",
       "      <td>5.7</td>\n",
       "      <td>0.03</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LRRC26</td>\n",
       "      <td>5.2</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MUC15</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.08</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GATAD2A</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.08</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MUT</td>\n",
       "      <td>4.7</td>\n",
       "      <td>0.08</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>490</th>\n",
       "      <td>CNTNAP2</td>\n",
       "      <td>2.6</td>\n",
       "      <td>0.25</td>\n",
       "      <td>523</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>491</th>\n",
       "      <td>hsa-mir-5693</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.25</td>\n",
       "      <td>525</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>492</th>\n",
       "      <td>hsa-mir-300</td>\n",
       "      <td>2.6</td>\n",
       "      <td>0.25</td>\n",
       "      <td>527</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>493</th>\n",
       "      <td>ZNF816-ZNF321P</td>\n",
       "      <td>2.6</td>\n",
       "      <td>0.23</td>\n",
       "      <td>531</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>494</th>\n",
       "      <td>hsa-mir-3129</td>\n",
       "      <td>2.6</td>\n",
       "      <td>0.25</td>\n",
       "      <td>552</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>495 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene Symbol  Score   FDR  GeCKO_number\n",
       "0            STXBP1    5.7  0.03             1\n",
       "1            LRRC26    5.2  0.08             3\n",
       "2             MUC15    5.0  0.08             4\n",
       "3           GATAD2A    5.0  0.08             5\n",
       "4               MUT    4.7  0.08             7\n",
       "..              ...    ...   ...           ...\n",
       "490         CNTNAP2    2.6  0.25           523\n",
       "491    hsa-mir-5693    0.3  0.25           525\n",
       "492     hsa-mir-300    2.6  0.25           527\n",
       "493  ZNF816-ZNF321P    2.6  0.23           531\n",
       "494    hsa-mir-3129    2.6  0.25           552\n",
       "\n",
       "[495 rows x 4 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# DL List of the SP1 candidat repressors\n",
    "SP1_repressors = pandas.read_csv(file_directory+'1) input/SP1_repressors.csv', sep = ',')\n",
    "SP1_repressors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "7670252b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>human</th>\n",
       "      <th>mouse</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1bg</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>A1cf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2LD1</td>\n",
       "      <td>A2ld1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2M</td>\n",
       "      <td>A2m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A4GALT</td>\n",
       "      <td>A4galt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15542</th>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>Zyg11a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15543</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>Zyg11b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15544</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>Zyx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15545</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>Zzef1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15546</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>Zzz3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15547 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        human   mouse\n",
       "0        A1BG    A1bg\n",
       "1        A1CF    A1cf\n",
       "2       A2LD1   A2ld1\n",
       "3         A2M     A2m\n",
       "4      A4GALT  A4galt\n",
       "...       ...     ...\n",
       "15542  ZYG11A  Zyg11a\n",
       "15543  ZYG11B  Zyg11b\n",
       "15544     ZYX     Zyx\n",
       "15545   ZZEF1   Zzef1\n",
       "15546    ZZZ3    Zzz3\n",
       "\n",
       "[15547 rows x 2 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# DL list human and mouse orthologs\n",
    "list_hum_mous = pandas.read_csv(file_directory+'human_mouse_1to1_orthologs.csv', sep = ',')\n",
    "list_hum_mous"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6d74df20",
   "metadata": {},
   "source": [
    "# Transform to have mice gene symbols "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "63f50869",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       human  Score   FDR  GeCKO_number     mouse\n",
      "0     STXBP1    5.7  0.03             1    Stxbp1\n",
      "1     LRRC26    5.2  0.08             3    Lrrc26\n",
      "2      MUC15    5.0  0.08             4     Muc15\n",
      "3    GATAD2A    5.0  0.08             5   Gatad2a\n",
      "4        MUT    4.7  0.08             7       Mut\n",
      "..       ...    ...   ...           ...       ...\n",
      "324    WNT11    2.6  0.25           507     Wnt11\n",
      "325   R3HDM1    2.6  0.25           508    R3hdm1\n",
      "326     RGMB    2.6  0.25           510      Rgmb\n",
      "327  C9orf78    2.6  0.25           522  BC005624\n",
      "328  CNTNAP2    2.6  0.25           523   Cntnap2\n",
      "\n",
      "[329 rows x 5 columns]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>Score</th>\n",
       "      <th>FDR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Stxbp1</td>\n",
       "      <td>5.7</td>\n",
       "      <td>0.03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Lrrc26</td>\n",
       "      <td>5.2</td>\n",
       "      <td>0.08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Muc15</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Gatad2a</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Mut</td>\n",
       "      <td>4.7</td>\n",
       "      <td>0.08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>324</th>\n",
       "      <td>Wnt11</td>\n",
       "      <td>2.6</td>\n",
       "      <td>0.25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>325</th>\n",
       "      <td>R3hdm1</td>\n",
       "      <td>2.6</td>\n",
       "      <td>0.25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>326</th>\n",
       "      <td>Rgmb</td>\n",
       "      <td>2.6</td>\n",
       "      <td>0.25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>327</th>\n",
       "      <td>BC005624</td>\n",
       "      <td>2.6</td>\n",
       "      <td>0.25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328</th>\n",
       "      <td>Cntnap2</td>\n",
       "      <td>2.6</td>\n",
       "      <td>0.25</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>329 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         gene  Score   FDR\n",
       "0      Stxbp1    5.7  0.03\n",
       "1      Lrrc26    5.2  0.08\n",
       "2       Muc15    5.0  0.08\n",
       "3     Gatad2a    5.0  0.08\n",
       "4         Mut    4.7  0.08\n",
       "..        ...    ...   ...\n",
       "324     Wnt11    2.6  0.25\n",
       "325    R3hdm1    2.6  0.25\n",
       "326      Rgmb    2.6  0.25\n",
       "327  BC005624    2.6  0.25\n",
       "328   Cntnap2    2.6  0.25\n",
       "\n",
       "[329 rows x 3 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#rename to be able to merge the two df\n",
    "SP1_repressors = SP1_repressors.rename(columns = {'Gene Symbol' : 'human'})\n",
    "\n",
    "# merge the SP1 df with the hum_mouse to have mouse gene symbols\n",
    "SP1_repressors = SP1_repressors.merge(list_hum_mous)\n",
    "print(SP1_repressors)\n",
    "\n",
    "SP1_repressors = SP1_repressors[['mouse', 'Score', 'FDR']]\n",
    "SP1_repressors = SP1_repressors.rename(columns = {'mouse' : 'gene'})\n",
    "SP1_repressors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "ea96436c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "number of Sp1 repressors in the RNAseq is 267 on a total of 329\n",
      "62 repressors are not in the RNAseq\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padj</th>\n",
       "      <th>lfcSE</th>\n",
       "      <th>Score</th>\n",
       "      <th>FDR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Creld2</td>\n",
       "      <td>-1.055765</td>\n",
       "      <td>4.374648e-44</td>\n",
       "      <td>2.014713e-41</td>\n",
       "      <td>0.075810</td>\n",
       "      <td>2.8</td>\n",
       "      <td>0.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Tfdp2</td>\n",
       "      <td>0.963359</td>\n",
       "      <td>3.843392e-42</td>\n",
       "      <td>1.511015e-39</td>\n",
       "      <td>0.070820</td>\n",
       "      <td>2.7</td>\n",
       "      <td>0.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Noc3l</td>\n",
       "      <td>-1.041630</td>\n",
       "      <td>1.709833e-27</td>\n",
       "      <td>2.375931e-25</td>\n",
       "      <td>0.095879</td>\n",
       "      <td>3.1</td>\n",
       "      <td>0.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Pias3</td>\n",
       "      <td>0.768008</td>\n",
       "      <td>1.526150e-26</td>\n",
       "      <td>1.983872e-24</td>\n",
       "      <td>0.072030</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Mns1</td>\n",
       "      <td>-1.103323</td>\n",
       "      <td>2.663300e-22</td>\n",
       "      <td>2.308050e-20</td>\n",
       "      <td>0.113597</td>\n",
       "      <td>3.9</td>\n",
       "      <td>0.16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>262</th>\n",
       "      <td>Hrk</td>\n",
       "      <td>-0.010509</td>\n",
       "      <td>9.783793e-01</td>\n",
       "      <td>9.890559e-01</td>\n",
       "      <td>0.387757</td>\n",
       "      <td>2.6</td>\n",
       "      <td>0.25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>263</th>\n",
       "      <td>Isca2</td>\n",
       "      <td>-0.002630</td>\n",
       "      <td>9.837204e-01</td>\n",
       "      <td>9.915327e-01</td>\n",
       "      <td>0.128887</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>264</th>\n",
       "      <td>Arfgap1</td>\n",
       "      <td>0.000418</td>\n",
       "      <td>9.951379e-01</td>\n",
       "      <td>9.976676e-01</td>\n",
       "      <td>0.068638</td>\n",
       "      <td>2.8</td>\n",
       "      <td>0.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>265</th>\n",
       "      <td>Gorab</td>\n",
       "      <td>0.000556</td>\n",
       "      <td>9.972183e-01</td>\n",
       "      <td>9.987600e-01</td>\n",
       "      <td>0.159352</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>266</th>\n",
       "      <td>Arl6ip5</td>\n",
       "      <td>0.000070</td>\n",
       "      <td>9.994792e-01</td>\n",
       "      <td>9.995569e-01</td>\n",
       "      <td>0.107616</td>\n",
       "      <td>2.6</td>\n",
       "      <td>0.25</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>267 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        gene  log2FoldChange        pvalue          padj     lfcSE  Score  \\\n",
       "0     Creld2       -1.055765  4.374648e-44  2.014713e-41  0.075810    2.8   \n",
       "1      Tfdp2        0.963359  3.843392e-42  1.511015e-39  0.070820    2.7   \n",
       "2      Noc3l       -1.041630  1.709833e-27  2.375931e-25  0.095879    3.1   \n",
       "3      Pias3        0.768008  1.526150e-26  1.983872e-24  0.072030    0.0   \n",
       "4       Mns1       -1.103323  2.663300e-22  2.308050e-20  0.113597    3.9   \n",
       "..       ...             ...           ...           ...       ...    ...   \n",
       "262      Hrk       -0.010509  9.783793e-01  9.890559e-01  0.387757    2.6   \n",
       "263    Isca2       -0.002630  9.837204e-01  9.915327e-01  0.128887    0.3   \n",
       "264  Arfgap1        0.000418  9.951379e-01  9.976676e-01  0.068638    2.8   \n",
       "265    Gorab        0.000556  9.972183e-01  9.987600e-01  0.159352    0.3   \n",
       "266  Arl6ip5        0.000070  9.994792e-01  9.995569e-01  0.107616    2.6   \n",
       "\n",
       "      FDR  \n",
       "0    0.23  \n",
       "1    0.23  \n",
       "2    0.22  \n",
       "3    0.24  \n",
       "4    0.16  \n",
       "..    ...  \n",
       "262  0.25  \n",
       "263  0.23  \n",
       "264  0.23  \n",
       "265  0.25  \n",
       "266  0.25  \n",
       "\n",
       "[267 rows x 7 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge the two tables and only keep the sp1_repressors in the RNAseq\n",
    "df_mNPC_sp1rep = pandas.merge(mouse_NPC_df, SP1_repressors, on= 'gene')\n",
    "\n",
    "print(f'number of Sp1 repressors in the RNAseq is {len(df_mNPC_sp1rep)} on a total of {len(SP1_repressors)}')\n",
    "print(f'{len(SP1_repressors) - len(df_mNPC_sp1rep)} repressors are not in the RNAseq')\n",
    "\n",
    "df_mNPC_sp1rep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "2dd973ec",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     gene  log2FoldChange        pvalue          padj    lfcSE\n",
      "40  Tfdp2        0.963359  3.843392e-42  1.511015e-39  0.07082\n",
      "    gene  log2FoldChange        pvalue          padj    lfcSE  Score   FDR\n",
      "1  Tfdp2        0.963359  3.843392e-42  1.511015e-39  0.07082    2.7  0.23\n"
     ]
    }
   ],
   "source": [
    "#control values did not change.\n",
    "print(mouse_NPC_df[mouse_NPC_df['gene'] == 'Tfdp2'])\n",
    "print(df_mNPC_sp1rep[df_mNPC_sp1rep['gene'] == 'Tfdp2'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "82f92a15",
   "metadata": {},
   "source": [
    "# TEST"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "5da9a939",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padj</th>\n",
       "      <th>lfcSE</th>\n",
       "      <th>Score</th>\n",
       "      <th>FDR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Creld2</td>\n",
       "      <td>-1.055765</td>\n",
       "      <td>4.374648e-44</td>\n",
       "      <td>2.014713e-41</td>\n",
       "      <td>0.075810</td>\n",
       "      <td>2.8</td>\n",
       "      <td>0.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Tfdp2</td>\n",
       "      <td>0.963359</td>\n",
       "      <td>3.843392e-42</td>\n",
       "      <td>1.511015e-39</td>\n",
       "      <td>0.070820</td>\n",
       "      <td>2.7</td>\n",
       "      <td>0.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Noc3l</td>\n",
       "      <td>-1.041630</td>\n",
       "      <td>1.709833e-27</td>\n",
       "      <td>2.375931e-25</td>\n",
       "      <td>0.095879</td>\n",
       "      <td>3.1</td>\n",
       "      <td>0.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Pias3</td>\n",
       "      <td>0.768008</td>\n",
       "      <td>1.526150e-26</td>\n",
       "      <td>1.983872e-24</td>\n",
       "      <td>0.072030</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Mns1</td>\n",
       "      <td>-1.103323</td>\n",
       "      <td>2.663300e-22</td>\n",
       "      <td>2.308050e-20</td>\n",
       "      <td>0.113597</td>\n",
       "      <td>3.9</td>\n",
       "      <td>0.16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>262</th>\n",
       "      <td>Hrk</td>\n",
       "      <td>-0.010509</td>\n",
       "      <td>9.783793e-01</td>\n",
       "      <td>9.890559e-01</td>\n",
       "      <td>0.387757</td>\n",
       "      <td>2.6</td>\n",
       "      <td>0.25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>263</th>\n",
       "      <td>Isca2</td>\n",
       "      <td>-0.002630</td>\n",
       "      <td>9.837204e-01</td>\n",
       "      <td>9.915327e-01</td>\n",
       "      <td>0.128887</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>264</th>\n",
       "      <td>Arfgap1</td>\n",
       "      <td>0.000418</td>\n",
       "      <td>9.951379e-01</td>\n",
       "      <td>9.976676e-01</td>\n",
       "      <td>0.068638</td>\n",
       "      <td>2.8</td>\n",
       "      <td>0.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>265</th>\n",
       "      <td>Gorab</td>\n",
       "      <td>0.000556</td>\n",
       "      <td>9.972183e-01</td>\n",
       "      <td>9.987600e-01</td>\n",
       "      <td>0.159352</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>266</th>\n",
       "      <td>Arl6ip5</td>\n",
       "      <td>0.000070</td>\n",
       "      <td>9.994792e-01</td>\n",
       "      <td>9.995569e-01</td>\n",
       "      <td>0.107616</td>\n",
       "      <td>2.6</td>\n",
       "      <td>0.25</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>267 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        gene  log2FoldChange        pvalue          padj     lfcSE  Score  \\\n",
       "0     Creld2       -1.055765  4.374648e-44  2.014713e-41  0.075810    2.8   \n",
       "1      Tfdp2        0.963359  3.843392e-42  1.511015e-39  0.070820    2.7   \n",
       "2      Noc3l       -1.041630  1.709833e-27  2.375931e-25  0.095879    3.1   \n",
       "3      Pias3        0.768008  1.526150e-26  1.983872e-24  0.072030    0.0   \n",
       "4       Mns1       -1.103323  2.663300e-22  2.308050e-20  0.113597    3.9   \n",
       "..       ...             ...           ...           ...       ...    ...   \n",
       "262      Hrk       -0.010509  9.783793e-01  9.890559e-01  0.387757    2.6   \n",
       "263    Isca2       -0.002630  9.837204e-01  9.915327e-01  0.128887    0.3   \n",
       "264  Arfgap1        0.000418  9.951379e-01  9.976676e-01  0.068638    2.8   \n",
       "265    Gorab        0.000556  9.972183e-01  9.987600e-01  0.159352    0.3   \n",
       "266  Arl6ip5        0.000070  9.994792e-01  9.995569e-01  0.107616    2.6   \n",
       "\n",
       "      FDR  \n",
       "0    0.23  \n",
       "1    0.23  \n",
       "2    0.22  \n",
       "3    0.24  \n",
       "4    0.16  \n",
       "..    ...  \n",
       "262  0.25  \n",
       "263  0.23  \n",
       "264  0.23  \n",
       "265  0.25  \n",
       "266  0.25  \n",
       "\n",
       "[267 rows x 7 columns]"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_mNPC_sp1rep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "cd7d17a3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Creld2</td>\n",
       "      <td>-1.055765</td>\n",
       "      <td>4.374648e-44</td>\n",
       "      <td>2.014713e-41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Tfdp2</td>\n",
       "      <td>0.963359</td>\n",
       "      <td>3.843392e-42</td>\n",
       "      <td>1.511015e-39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Noc3l</td>\n",
       "      <td>-1.041630</td>\n",
       "      <td>1.709833e-27</td>\n",
       "      <td>2.375931e-25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Pias3</td>\n",
       "      <td>0.768008</td>\n",
       "      <td>1.526150e-26</td>\n",
       "      <td>1.983872e-24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Mns1</td>\n",
       "      <td>-1.103323</td>\n",
       "      <td>2.663300e-22</td>\n",
       "      <td>2.308050e-20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>262</th>\n",
       "      <td>Hrk</td>\n",
       "      <td>-0.010509</td>\n",
       "      <td>9.783793e-01</td>\n",
       "      <td>9.890559e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>263</th>\n",
       "      <td>Isca2</td>\n",
       "      <td>-0.002630</td>\n",
       "      <td>9.837204e-01</td>\n",
       "      <td>9.915327e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>264</th>\n",
       "      <td>Arfgap1</td>\n",
       "      <td>0.000418</td>\n",
       "      <td>9.951379e-01</td>\n",
       "      <td>9.976676e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>265</th>\n",
       "      <td>Gorab</td>\n",
       "      <td>0.000556</td>\n",
       "      <td>9.972183e-01</td>\n",
       "      <td>9.987600e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>266</th>\n",
       "      <td>Arl6ip5</td>\n",
       "      <td>0.000070</td>\n",
       "      <td>9.994792e-01</td>\n",
       "      <td>9.995569e-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>267 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        gene  log2FoldChange        pvalue          padj\n",
       "0     Creld2       -1.055765  4.374648e-44  2.014713e-41\n",
       "1      Tfdp2        0.963359  3.843392e-42  1.511015e-39\n",
       "2      Noc3l       -1.041630  1.709833e-27  2.375931e-25\n",
       "3      Pias3        0.768008  1.526150e-26  1.983872e-24\n",
       "4       Mns1       -1.103323  2.663300e-22  2.308050e-20\n",
       "..       ...             ...           ...           ...\n",
       "262      Hrk       -0.010509  9.783793e-01  9.890559e-01\n",
       "263    Isca2       -0.002630  9.837204e-01  9.915327e-01\n",
       "264  Arfgap1        0.000418  9.951379e-01  9.976676e-01\n",
       "265    Gorab        0.000556  9.972183e-01  9.987600e-01\n",
       "266  Arl6ip5        0.000070  9.994792e-01  9.995569e-01\n",
       "\n",
       "[267 rows x 4 columns]"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_mNPC_sp1rep = df_mNPC_sp1rep[['gene', 'log2FoldChange', 'pvalue', 'padj']]\n",
    "df_mNPC_sp1rep = df_mNPC_sp1rep.set_index('gene')\n",
    "\n",
    "df_mNPC_sp1rep.to_csv(file_directory+'1) input/from_output/df_mNPC_sp1rep.tsv',sep = \"\\t\")\n",
    "df_mNPC_sp1rep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "6154915e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(267, 3)\n"
     ]
    }
   ],
   "source": [
    "adata  = scanpy.read_csv(file_directory+'1) input/from_output/df_mNPC_sp1rep.tsv', delimiter='\\t')\n",
    "\n",
    "print(adata.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "dd821249",
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'DataFrame' object has no attribute 'is_view'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[1;32m~\\AppData\\Local\\Temp\\ipykernel_16996\\2541565481.py\u001b[0m in \u001b[0;36m?\u001b[1;34m()\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[0mnei\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;36m5\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 2\u001b[1;33m \u001b[0mscanpy\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mpp\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mneighbors\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mdf_mNPC_sp1rep\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mn_neighbors\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mnei\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mn_pcs\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m100\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;32m~\\anaconda3\\Lib\\site-packages\\scanpy\\neighbors\\__init__.py\u001b[0m in \u001b[0;36m?\u001b[1;34m(adata, n_neighbors, n_pcs, use_rep, knn, random_state, method, metric, metric_kwds, key_added, copy)\u001b[0m\n\u001b[0;32m    141\u001b[0m     \u001b[1;32mand\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mlater\u001b[0m \u001b[0mversions\u001b[0m \u001b[0mit\u001b[0m \u001b[0mwill\u001b[0m \u001b[0mbecome\u001b[0m \u001b[0ma\u001b[0m \u001b[0mhard\u001b[0m \u001b[0mdependency\u001b[0m\u001b[1;33m.\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    142\u001b[0m     \"\"\"\n\u001b[0;32m    143\u001b[0m     \u001b[0mstart\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mlogg\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0minfo\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'computing neighbors'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    144\u001b[0m     \u001b[0madata\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0madata\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcopy\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mif\u001b[0m \u001b[0mcopy\u001b[0m \u001b[1;32melse\u001b[0m \u001b[0madata\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 145\u001b[1;33m     \u001b[1;32mif\u001b[0m \u001b[0madata\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mis_view\u001b[0m\u001b[1;33m:\u001b[0m  \u001b[1;31m# we shouldn't need this here...\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    146\u001b[0m         \u001b[0madata\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_init_as_actual\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0madata\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcopy\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    147\u001b[0m     \u001b[0mneighbors\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mNeighbors\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0madata\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    148\u001b[0m     neighbors.compute_neighbors(\n",
      "\u001b[1;32m~\\anaconda3\\Lib\\site-packages\\pandas\\core\\generic.py\u001b[0m in \u001b[0;36m?\u001b[1;34m(self, name)\u001b[0m\n\u001b[0;32m   5985\u001b[0m             \u001b[1;32mand\u001b[0m \u001b[0mname\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_accessors\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   5986\u001b[0m             \u001b[1;32mand\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_info_axis\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_can_hold_identifiers_and_holds_name\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mname\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   5987\u001b[0m         ):\n\u001b[0;32m   5988\u001b[0m             \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mname\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 5989\u001b[1;33m         \u001b[1;32mreturn\u001b[0m \u001b[0mobject\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m__getattribute__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mname\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;31mAttributeError\u001b[0m: 'DataFrame' object has no attribute 'is_view'"
     ]
    }
   ],
   "source": [
    "nei = 5\n",
    "scanpy.pp.neighbors(df_mNPC_sp1rep, n_neighbors=nei, n_pcs=100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "717cd0fe",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unexpected exception formatting exception. Falling back to standard exception\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\kja11\\anaconda3\\Lib\\site-packages\\IPython\\core\\magics\\execution.py\", line 1340, in time\n",
      "    exec(code, glob, local_ns)\n",
      "  File \"<timed exec>\", line 4, in <module>\n",
      "  File \"C:\\Users\\kja11\\anaconda3\\Lib\\site-packages\\scanpy\\neighbors\\__init__.py\", line 145, in neighbors\n",
      "    if adata.is_view:  # we shouldn't need this here...\n",
      "       ^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\kja11\\anaconda3\\Lib\\site-packages\\pandas\\core\\generic.py\", line 5989, in __getattr__\n",
      "    return object.__getattribute__(self, name)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "AttributeError: 'DataFrame' object has no attribute 'is_view'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\kja11\\anaconda3\\Lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2120, in showtraceback\n",
      "    stb = self.InteractiveTB.structured_traceback(\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\kja11\\anaconda3\\Lib\\site-packages\\IPython\\core\\ultratb.py\", line 1435, in structured_traceback\n",
      "    return FormattedTB.structured_traceback(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\kja11\\anaconda3\\Lib\\site-packages\\IPython\\core\\ultratb.py\", line 1326, in structured_traceback\n",
      "    return VerboseTB.structured_traceback(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\kja11\\anaconda3\\Lib\\site-packages\\IPython\\core\\ultratb.py\", line 1173, in structured_traceback\n",
      "    formatted_exception = self.format_exception_as_a_whole(etype, evalue, etb, number_of_lines_of_context,\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\kja11\\anaconda3\\Lib\\site-packages\\IPython\\core\\ultratb.py\", line 1063, in format_exception_as_a_whole\n",
      "    self.get_records(etb, number_of_lines_of_context, tb_offset) if etb else []\n",
      "    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\kja11\\anaconda3\\Lib\\site-packages\\IPython\\core\\ultratb.py\", line 1155, in get_records\n",
      "    FrameInfo(\n",
      "  File \"C:\\Users\\kja11\\anaconda3\\Lib\\site-packages\\IPython\\core\\ultratb.py\", line 780, in __init__\n",
      "    ix = inspect.getsourcelines(frame)\n",
      "         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\kja11\\anaconda3\\Lib\\inspect.py\", line 1244, in getsourcelines\n",
      "    lines, lnum = findsource(object)\n",
      "                  ^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\kja11\\anaconda3\\Lib\\inspect.py\", line 1081, in findsource\n",
      "    raise OSError('could not get source code')\n",
      "OSError: could not get source code\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "\n",
    "nei = 5\n",
    "dist = 0.2\n",
    "spre = 10\n",
    "scanpy.pp.neighbors(df_final, n_neighbors=nei, n_pcs=100)\n",
    "scanpy.tl.leiden(adata)\n",
    "scanpy.tl.paga(adata)\n",
    "scanpy.pl.paga(adata, plot=False)\n",
    "scanpy.tl.umap(adata, min_dist=dist, spread=spre, init_pos='paga')      \n",
    "scanpy.pl.umap(adata, color='leiden', palette=two_hundred_colors, legend_loc=None, save='_all')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4c18556f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
