{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "811bf9d9",
   "metadata": {},
   "source": [
    "# Import, options"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c7a0dcfb",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy, matplotlib, pandas, scanpy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "01ebcba6",
   "metadata": {},
   "outputs": [],
   "source": [
    "matplotlib.rcParams.update({'font.size':20, 'xtick.labelsize':20, 'ytick.labelsize':20, \n",
    "                            'axes.grid' : True, 'grid.alpha': 0.5, 'grid.linestyle' : ':',\n",
    "                            'figure.figsize':(8, 5), 'svg.fonttype' : 'none'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "9b6430a8",
   "metadata": {},
   "outputs": [],
   "source": [
    "file_directory = \"/Users/kja11/OneDrive - Menntaský/PostDoc_Hypothermia/in_silico/Python/\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a9797af2",
   "metadata": {},
   "source": [
    "# Data Downloads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "1cb616f4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(16062, 15) \n",
      " Index(['Row.names', 'baseMean', 'log2FoldChange', 'lfcSE', 'stat', 'pvalue',\n",
      "       'padj', 'gene', '37_degrees', '37_degrees.1', '37_degrees.2',\n",
      "       '32_degrees', '32_degrees.1', '32_degrees.2', '32_degrees.3'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>32_degrees</th>\n",
       "      <th>32_degrees.1</th>\n",
       "      <th>32_degrees.2</th>\n",
       "      <th>32_degrees.3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Cirbp</td>\n",
       "      <td>6117.014966</td>\n",
       "      <td>5155.322063</td>\n",
       "      <td>5886.014969</td>\n",
       "      <td>5396.949830</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Xbp1</td>\n",
       "      <td>683.013572</td>\n",
       "      <td>668.546151</td>\n",
       "      <td>700.131803</td>\n",
       "      <td>612.231119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Hspa8</td>\n",
       "      <td>46940.696119</td>\n",
       "      <td>43593.055672</td>\n",
       "      <td>45443.791959</td>\n",
       "      <td>41847.851131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Dnaja3</td>\n",
       "      <td>1225.709230</td>\n",
       "      <td>1163.293883</td>\n",
       "      <td>1185.274805</td>\n",
       "      <td>1204.699735</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Tra2a</td>\n",
       "      <td>652.603893</td>\n",
       "      <td>620.523550</td>\n",
       "      <td>590.851333</td>\n",
       "      <td>453.756987</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16057</th>\n",
       "      <td>Pex2</td>\n",
       "      <td>1629.899000</td>\n",
       "      <td>1835.107000</td>\n",
       "      <td>1833.215000</td>\n",
       "      <td>1462.453000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16058</th>\n",
       "      <td>Ank3</td>\n",
       "      <td>990.711700</td>\n",
       "      <td>794.611200</td>\n",
       "      <td>910.848200</td>\n",
       "      <td>715.005000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16059</th>\n",
       "      <td>Hmgcl</td>\n",
       "      <td>1115.568000</td>\n",
       "      <td>995.605900</td>\n",
       "      <td>1094.933000</td>\n",
       "      <td>1095.132000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16060</th>\n",
       "      <td>Arl6ip5</td>\n",
       "      <td>1355.401000</td>\n",
       "      <td>1512.990000</td>\n",
       "      <td>1346.521000</td>\n",
       "      <td>1217.732000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16061</th>\n",
       "      <td>Wdr7</td>\n",
       "      <td>162.290500</td>\n",
       "      <td>100.989900</td>\n",
       "      <td>74.985460</td>\n",
       "      <td>167.580800</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16062 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          gene    32_degrees  32_degrees.1  32_degrees.2  32_degrees.3\n",
       "0        Cirbp   6117.014966   5155.322063   5886.014969   5396.949830\n",
       "1         Xbp1    683.013572    668.546151    700.131803    612.231119\n",
       "2        Hspa8  46940.696119  43593.055672  45443.791959  41847.851131\n",
       "3       Dnaja3   1225.709230   1163.293883   1185.274805   1204.699735\n",
       "4        Tra2a    652.603893    620.523550    590.851333    453.756987\n",
       "...        ...           ...           ...           ...           ...\n",
       "16057     Pex2   1629.899000   1835.107000   1833.215000   1462.453000\n",
       "16058     Ank3    990.711700    794.611200    910.848200    715.005000\n",
       "16059    Hmgcl   1115.568000    995.605900   1094.933000   1095.132000\n",
       "16060  Arl6ip5   1355.401000   1512.990000   1346.521000   1217.732000\n",
       "16061     Wdr7    162.290500    100.989900     74.985460    167.580800\n",
       "\n",
       "[16062 rows x 5 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#data mouse_NPC\n",
    "mouse_NPC_df = pandas.read_csv(file_directory+'1) input/mNPC_all_detected_genes_DEseq.csv',\n",
    "                               sep = ',')\n",
    "print(mouse_NPC_df.shape, '\\n', mouse_NPC_df.columns)\n",
    "\n",
    "mouse_NPC_df = mouse_NPC_df[['gene', '32_degrees', '32_degrees.1', '32_degrees.2', '32_degrees.3']]\n",
    "mouse_NPC_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "15c8f337",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>STXBP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LRRC26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MUC15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GATAD2A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MUT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>490</th>\n",
       "      <td>CNTNAP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>491</th>\n",
       "      <td>hsa-mir-5693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>492</th>\n",
       "      <td>hsa-mir-300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>493</th>\n",
       "      <td>ZNF816-ZNF321P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>494</th>\n",
       "      <td>hsa-mir-3129</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>495 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene Symbol\n",
       "0            STXBP1\n",
       "1            LRRC26\n",
       "2             MUC15\n",
       "3           GATAD2A\n",
       "4               MUT\n",
       "..              ...\n",
       "490         CNTNAP2\n",
       "491    hsa-mir-5693\n",
       "492     hsa-mir-300\n",
       "493  ZNF816-ZNF321P\n",
       "494    hsa-mir-3129\n",
       "\n",
       "[495 rows x 1 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# DL List of the SP1 candidat repressors\n",
    "SP1_repressors = pandas.read_csv(file_directory+'1) input/SP1_repressors.csv', sep = ',')\n",
    "SP1_repressors = SP1_repressors[['Gene Symbol']]\n",
    "SP1_repressors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "7670252b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>human</th>\n",
       "      <th>mouse</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1bg</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>A1cf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2LD1</td>\n",
       "      <td>A2ld1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2M</td>\n",
       "      <td>A2m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A4GALT</td>\n",
       "      <td>A4galt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15542</th>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>Zyg11a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15543</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>Zyg11b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15544</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>Zyx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15545</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>Zzef1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15546</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>Zzz3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15547 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        human   mouse\n",
       "0        A1BG    A1bg\n",
       "1        A1CF    A1cf\n",
       "2       A2LD1   A2ld1\n",
       "3         A2M     A2m\n",
       "4      A4GALT  A4galt\n",
       "...       ...     ...\n",
       "15542  ZYG11A  Zyg11a\n",
       "15543  ZYG11B  Zyg11b\n",
       "15544     ZYX     Zyx\n",
       "15545   ZZEF1   Zzef1\n",
       "15546    ZZZ3    Zzz3\n",
       "\n",
       "[15547 rows x 2 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# DL list human and mouse orthologs\n",
    "list_hum_mous = pandas.read_csv(file_directory+'human_mouse_1to1_orthologs.csv', sep = ',')\n",
    "list_hum_mous"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6d74df20",
   "metadata": {},
   "source": [
    "# Transform to have mice gene symbols "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "63f50869",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       human     mouse\n",
      "0     STXBP1    Stxbp1\n",
      "1     LRRC26    Lrrc26\n",
      "2      MUC15     Muc15\n",
      "3    GATAD2A   Gatad2a\n",
      "4        MUT       Mut\n",
      "..       ...       ...\n",
      "324    WNT11     Wnt11\n",
      "325   R3HDM1    R3hdm1\n",
      "326     RGMB      Rgmb\n",
      "327  C9orf78  BC005624\n",
      "328  CNTNAP2   Cntnap2\n",
      "\n",
      "[329 rows x 2 columns]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Stxbp1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Lrrc26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Muc15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Gatad2a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Mut</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>324</th>\n",
       "      <td>Wnt11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>325</th>\n",
       "      <td>R3hdm1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>326</th>\n",
       "      <td>Rgmb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>327</th>\n",
       "      <td>BC005624</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328</th>\n",
       "      <td>Cntnap2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>329 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         gene\n",
       "0      Stxbp1\n",
       "1      Lrrc26\n",
       "2       Muc15\n",
       "3     Gatad2a\n",
       "4         Mut\n",
       "..        ...\n",
       "324     Wnt11\n",
       "325    R3hdm1\n",
       "326      Rgmb\n",
       "327  BC005624\n",
       "328   Cntnap2\n",
       "\n",
       "[329 rows x 1 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#rename to be able to merge the two df\n",
    "SP1_repressors = SP1_repressors.rename(columns = {'Gene Symbol' : 'human'})\n",
    "\n",
    "# merge the SP1 df with the hum_mouse to have mouse gene symbols\n",
    "SP1_repressors = SP1_repressors.merge(list_hum_mous)\n",
    "print(SP1_repressors)\n",
    "\n",
    "SP1_repressors = SP1_repressors[['mouse']]\n",
    "SP1_repressors = SP1_repressors.rename(columns = {'mouse' : 'gene'})\n",
    "SP1_repressors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "ea96436c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "number of Sp1 repressors in the RNAseq is 267 on a total of 329\n",
      "62 repressors are not in the RNAseq\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>32_degrees</th>\n",
       "      <th>32_degrees.1</th>\n",
       "      <th>32_degrees.2</th>\n",
       "      <th>32_degrees.3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Creld2</td>\n",
       "      <td>751.602695</td>\n",
       "      <td>748.099155</td>\n",
       "      <td>778.773074</td>\n",
       "      <td>801.580112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Tfdp2</td>\n",
       "      <td>3349.830478</td>\n",
       "      <td>3247.058555</td>\n",
       "      <td>3030.374898</td>\n",
       "      <td>3307.762556</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Noc3l</td>\n",
       "      <td>827.786899</td>\n",
       "      <td>853.633273</td>\n",
       "      <td>754.984167</td>\n",
       "      <td>665.868631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Pias3</td>\n",
       "      <td>1454.152781</td>\n",
       "      <td>1520.922170</td>\n",
       "      <td>1563.303656</td>\n",
       "      <td>1484.663052</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Mns1</td>\n",
       "      <td>975.424977</td>\n",
       "      <td>783.570507</td>\n",
       "      <td>944.911361</td>\n",
       "      <td>746.454093</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene   32_degrees  32_degrees.1  32_degrees.2  32_degrees.3\n",
       "0  Creld2   751.602695    748.099155    778.773074    801.580112\n",
       "1   Tfdp2  3349.830478   3247.058555   3030.374898   3307.762556\n",
       "2   Noc3l   827.786899    853.633273    754.984167    665.868631\n",
       "3   Pias3  1454.152781   1520.922170   1563.303656   1484.663052\n",
       "4    Mns1   975.424977    783.570507    944.911361    746.454093"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge the two tables and only keep the sp1_repressors in the RNAseq\n",
    "df_mNPC_sp1rep = pandas.merge(mouse_NPC_df, SP1_repressors, on= 'gene')\n",
    "\n",
    "print(f'number of Sp1 repressors in the RNAseq is {len(df_mNPC_sp1rep)} on a total of {len(SP1_repressors)}')\n",
    "print(f'{len(SP1_repressors) - len(df_mNPC_sp1rep)} repressors are not in the RNAseq')\n",
    "\n",
    "df_mNPC_sp1rep.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "2dd973ec",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     gene   32_degrees  32_degrees.1  32_degrees.2  32_degrees.3\n",
      "40  Tfdp2  3349.830478   3247.058555   3030.374898   3307.762556\n",
      "    gene   32_degrees  32_degrees.1  32_degrees.2  32_degrees.3\n",
      "1  Tfdp2  3349.830478   3247.058555   3030.374898   3307.762556\n"
     ]
    }
   ],
   "source": [
    "#control values did not change.\n",
    "print(mouse_NPC_df[mouse_NPC_df['gene'] == 'Tfdp2'])\n",
    "print(df_mNPC_sp1rep[df_mNPC_sp1rep['gene'] == 'Tfdp2'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "82f92a15",
   "metadata": {},
   "source": [
    "# UMAP from louvain algorithm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "cd7d17a3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>32_degrees</th>\n",
       "      <th>32_degrees.1</th>\n",
       "      <th>32_degrees.2</th>\n",
       "      <th>32_degrees.3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Creld2</th>\n",
       "      <td>751.602695</td>\n",
       "      <td>748.099155</td>\n",
       "      <td>778.773074</td>\n",
       "      <td>801.580112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Tfdp2</th>\n",
       "      <td>3349.830478</td>\n",
       "      <td>3247.058555</td>\n",
       "      <td>3030.374898</td>\n",
       "      <td>3307.762556</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Noc3l</th>\n",
       "      <td>827.786899</td>\n",
       "      <td>853.633273</td>\n",
       "      <td>754.984167</td>\n",
       "      <td>665.868631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pias3</th>\n",
       "      <td>1454.152781</td>\n",
       "      <td>1520.922170</td>\n",
       "      <td>1563.303656</td>\n",
       "      <td>1484.663052</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mns1</th>\n",
       "      <td>975.424977</td>\n",
       "      <td>783.570507</td>\n",
       "      <td>944.911361</td>\n",
       "      <td>746.454093</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         32_degrees  32_degrees.1  32_degrees.2  32_degrees.3\n",
       "gene                                                         \n",
       "Creld2   751.602695    748.099155    778.773074    801.580112\n",
       "Tfdp2   3349.830478   3247.058555   3030.374898   3307.762556\n",
       "Noc3l    827.786899    853.633273    754.984167    665.868631\n",
       "Pias3   1454.152781   1520.922170   1563.303656   1484.663052\n",
       "Mns1     975.424977    783.570507    944.911361    746.454093"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Save the file to .csv\n",
    "df_mNPC_sp1rep = df_mNPC_sp1rep.set_index('gene')\n",
    "\n",
    "df_mNPC_sp1rep.to_csv(file_directory+'1) input/from_output/df_mNPC_sp1rep.tsv',sep = \"\\t\")\n",
    "df_mNPC_sp1rep.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "358541e3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(267, 4)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "AnnData object with n_obs × n_vars = 267 × 4"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Open the saved file with scanpy\n",
    "adata  = scanpy.read_csv(file_directory+'1) input/from_output/df_mNPC_sp1rep.tsv', delimiter='\\t')\n",
    "\n",
    "#adata for anotated data\n",
    "print(adata.shape)\n",
    "adata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "24c3f098",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "genes in Smyd5 group: Index(['Myo1a', 'Arhgap25', 'Cyp4x1', 'Tal2', 'Adra2b', 'Tyr', 'Rasd1', 'Bgn',\n",
      "       'Rho', 'Col14a1', 'Cntnap2', 'B4galnt3', 'Kcnk12', 'Chmp4c', 'Scube2',\n",
      "       'Grhl3', 'Tulp1', 'D130043K22Rik', 'Muc15', 'Dlec1', 'Nxnl1', 'Igf2bp1',\n",
      "       'Calhm2', 'Prlr', 'Cd274', 'C1rl', 'Gjd4', 'Mapk15', 'Cyp2r1', 'Hhex',\n",
      "       'Pon3'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "# yellow is group 0. Get all the gene in group 0\n",
    "group_0_elements = adata.obs[adata.obs['louvain'] == '0'].index\n",
    "\n",
    "print(f'genes in Smyd5 group: {group_0_elements}')\n",
    "\n",
    "f = open(file_directory+'3) output/gene_in_smyd5_group.txt', 'w')\n",
    "for element in group_0_elements:\n",
    "  f.write('{}\\n'.format(element))\n",
    "\n",
    "f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "0acb82db",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Myo1a', 'Arhgap25', 'Cyp4x1', 'Tal2', 'Adra2b', 'Tyr', 'Rasd1', 'Bgn',\n",
       "       'Rho', 'Col14a1', 'Cntnap2', 'B4galnt3', 'Kcnk12', 'Chmp4c', 'Scube2',\n",
       "       'Grhl3', 'Tulp1', 'D130043K22Rik', 'Muc15', 'Dlec1', 'Nxnl1', 'Igf2bp1',\n",
       "       'Calhm2', 'Prlr', 'Cd274', 'C1rl', 'Gjd4', 'Mapk15', 'Cyp2r1', 'Hhex',\n",
       "       'Pon3'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_0_elements"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
