{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "19069ee0",
   "metadata": {},
   "source": [
    "# Import, options"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "2adb646b",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy, seaborn, pandas\n",
    "\n",
    "import pybiomart\n",
    "from pydeseq2.dds import DeseqDataSet\n",
    "from pydeseq2.ds import DeseqStats\n",
    "\n",
    "import matplotlib, matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "0c188f4e",
   "metadata": {},
   "outputs": [],
   "source": [
    "matplotlib.rcParams.update({'svg.fonttype' : 'none'})\n",
    "#'font.size':20, 'xtick.labelsize':20, 'ytick.labelsize':20,'axes.grid' : True, 'grid.alpha': 0.5, 'grid.linestyle' : ':',                            'figure.figsize':(8, 5), "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "6ae19579",
   "metadata": {},
   "outputs": [],
   "source": [
    "file_directory = \"/Users/kja11/OneDrive - Menntaský/PostDoc_Hypothermia/in_silico/Python/\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8e0197ea",
   "metadata": {},
   "source": [
    "# Data Download"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "1268085d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(22752, 8) \n",
      " Index(['gene_id', 'gene_name', 's37rep1_sum', 's37rep2_sum', 's37rep3_sum',\n",
      "       's32rep1_sum', 's32rep2_sum', 's32rep3_sum'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>s37rep1_sum</th>\n",
       "      <th>s37rep2_sum</th>\n",
       "      <th>s37rep3_sum</th>\n",
       "      <th>s32rep1_sum</th>\n",
       "      <th>s32rep2_sum</th>\n",
       "      <th>s32rep3_sum</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>TSPAN6</td>\n",
       "      <td>1669</td>\n",
       "      <td>1746</td>\n",
       "      <td>1154</td>\n",
       "      <td>1319</td>\n",
       "      <td>1492</td>\n",
       "      <td>1265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000005</td>\n",
       "      <td>TNMD</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000419</td>\n",
       "      <td>DPM1</td>\n",
       "      <td>5097</td>\n",
       "      <td>5407</td>\n",
       "      <td>3552</td>\n",
       "      <td>4640</td>\n",
       "      <td>5216</td>\n",
       "      <td>4412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000457</td>\n",
       "      <td>SCYL3</td>\n",
       "      <td>158</td>\n",
       "      <td>173</td>\n",
       "      <td>136</td>\n",
       "      <td>113</td>\n",
       "      <td>97</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000000460</td>\n",
       "      <td>C1orf112</td>\n",
       "      <td>875</td>\n",
       "      <td>850</td>\n",
       "      <td>650</td>\n",
       "      <td>481</td>\n",
       "      <td>635</td>\n",
       "      <td>618</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22747</th>\n",
       "      <td>ENSG00000292361</td>\n",
       "      <td>CD99P1</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>36</td>\n",
       "      <td>37</td>\n",
       "      <td>22</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22748</th>\n",
       "      <td>ENSG00000292366</td>\n",
       "      <td>VAMP7</td>\n",
       "      <td>568</td>\n",
       "      <td>459</td>\n",
       "      <td>279</td>\n",
       "      <td>459</td>\n",
       "      <td>412</td>\n",
       "      <td>303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22749</th>\n",
       "      <td>ENSG00000292371</td>\n",
       "      <td>DDX11L16</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22750</th>\n",
       "      <td>ENSG00000292372</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>486</td>\n",
       "      <td>555</td>\n",
       "      <td>515</td>\n",
       "      <td>528</td>\n",
       "      <td>464</td>\n",
       "      <td>305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22751</th>\n",
       "      <td>ENSG00000292373</td>\n",
       "      <td>IL9R</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>22752 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               gene_id gene_name  s37rep1_sum  s37rep2_sum  s37rep3_sum  \\\n",
       "0      ENSG00000000003    TSPAN6         1669         1746         1154   \n",
       "1      ENSG00000000005      TNMD            3            0            0   \n",
       "2      ENSG00000000419      DPM1         5097         5407         3552   \n",
       "3      ENSG00000000457     SCYL3          158          173          136   \n",
       "4      ENSG00000000460  C1orf112          875          850          650   \n",
       "...                ...       ...          ...          ...          ...   \n",
       "22747  ENSG00000292361    CD99P1           30           23           36   \n",
       "22748  ENSG00000292366     VAMP7          568          459          279   \n",
       "22749  ENSG00000292371  DDX11L16            1            0            0   \n",
       "22750  ENSG00000292372    WASH6P          486          555          515   \n",
       "22751  ENSG00000292373      IL9R            6            0            0   \n",
       "\n",
       "       s32rep1_sum  s32rep2_sum  s32rep3_sum  \n",
       "0             1319         1492         1265  \n",
       "1                1            0            2  \n",
       "2             4640         5216         4412  \n",
       "3              113           97          145  \n",
       "4              481          635          618  \n",
       "...            ...          ...          ...  \n",
       "22747           37           22           36  \n",
       "22748          459          412          303  \n",
       "22749            0            0            0  \n",
       "22750          528          464          305  \n",
       "22751           13           10            0  \n",
       "\n",
       "[22752 rows x 8 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#data DEseq HEK293 significant ones\n",
    "df = pandas.read_csv(file_directory+'1) input/RNAseq/DESeq_HEK293_temp_v2.tsv',\n",
    "                               sep = '\\t')\n",
    "print(df.shape, '\\n', df.columns)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "cade9efa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>Gene type</th>\n",
       "      <th>Gene description</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000210049</th>\n",
       "      <td>MT-TF</td>\n",
       "      <td>Mt_tRNA</td>\n",
       "      <td>mitochondrially encoded tRNA-Phe (UUU/C) [Sour...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000211459</th>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>Mt_rRNA</td>\n",
       "      <td>mitochondrially encoded 12S rRNA [Source:HGNC ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                gene_name Gene type  \\\n",
       "gene_id                               \n",
       "ENSG00000210049     MT-TF   Mt_tRNA   \n",
       "ENSG00000211459   MT-RNR1   Mt_rRNA   \n",
       "\n",
       "                                                  Gene description  \n",
       "gene_id                                                             \n",
       "ENSG00000210049  mitochondrially encoded tRNA-Phe (UUU/C) [Sour...  \n",
       "ENSG00000211459  mitochondrially encoded 12S rRNA [Source:HGNC ...  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Import data from hsapiens_gene_ensembl\n",
    "dataset = pybiomart.Dataset(name='hsapiens_gene_ensembl', host='http://www.ensembl.org')\n",
    "annotation = dataset.query(attributes=['external_gene_name', 'ensembl_gene_id', 'gene_biotype'\n",
    "                                       ,'description'])\n",
    "\n",
    "annotation = annotation.rename(columns = {'Gene stable ID' : 'gene_id',\n",
    "                                          'Gene name': 'gene_name'})\n",
    "\n",
    "annotation.set_index('gene_id', inplace=True)\n",
    "annotation.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4d1864fb",
   "metadata": {},
   "source": [
    "## Data transformation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "8c1fa80a",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>s37rep1_sum</th>\n",
       "      <th>s37rep2_sum</th>\n",
       "      <th>s37rep3_sum</th>\n",
       "      <th>s32rep1_sum</th>\n",
       "      <th>s32rep2_sum</th>\n",
       "      <th>s32rep3_sum</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000000003</th>\n",
       "      <td>1669</td>\n",
       "      <td>1746</td>\n",
       "      <td>1154</td>\n",
       "      <td>1319</td>\n",
       "      <td>1492</td>\n",
       "      <td>1265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000005</th>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000419</th>\n",
       "      <td>5097</td>\n",
       "      <td>5407</td>\n",
       "      <td>3552</td>\n",
       "      <td>4640</td>\n",
       "      <td>5216</td>\n",
       "      <td>4412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000457</th>\n",
       "      <td>158</td>\n",
       "      <td>173</td>\n",
       "      <td>136</td>\n",
       "      <td>113</td>\n",
       "      <td>97</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000460</th>\n",
       "      <td>875</td>\n",
       "      <td>850</td>\n",
       "      <td>650</td>\n",
       "      <td>481</td>\n",
       "      <td>635</td>\n",
       "      <td>618</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 s37rep1_sum  s37rep2_sum  s37rep3_sum  s32rep1_sum  \\\n",
       "gene_id                                                               \n",
       "ENSG00000000003         1669         1746         1154         1319   \n",
       "ENSG00000000005            3            0            0            1   \n",
       "ENSG00000000419         5097         5407         3552         4640   \n",
       "ENSG00000000457          158          173          136          113   \n",
       "ENSG00000000460          875          850          650          481   \n",
       "\n",
       "                 s32rep2_sum  s32rep3_sum  \n",
       "gene_id                                    \n",
       "ENSG00000000003         1492         1265  \n",
       "ENSG00000000005            0            2  \n",
       "ENSG00000000419         5216         4412  \n",
       "ENSG00000000457           97          145  \n",
       "ENSG00000000460          635          618  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Organize the df\n",
    "df.set_index('gene_id', drop=True, inplace=True)\n",
    "df = df.drop(['gene_name'], axis=1)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "e5b90482",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "On 22752, 7429 transcripts do not exceed 10 counts\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>gene_id</th>\n",
       "      <th>ENSG00000000003</th>\n",
       "      <th>ENSG00000000419</th>\n",
       "      <th>ENSG00000000457</th>\n",
       "      <th>ENSG00000000460</th>\n",
       "      <th>ENSG00000000971</th>\n",
       "      <th>ENSG00000001036</th>\n",
       "      <th>ENSG00000001084</th>\n",
       "      <th>ENSG00000001167</th>\n",
       "      <th>ENSG00000001460</th>\n",
       "      <th>ENSG00000001461</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSG00000292343</th>\n",
       "      <th>ENSG00000292344</th>\n",
       "      <th>ENSG00000292345</th>\n",
       "      <th>ENSG00000292348</th>\n",
       "      <th>ENSG00000292357</th>\n",
       "      <th>ENSG00000292358</th>\n",
       "      <th>ENSG00000292361</th>\n",
       "      <th>ENSG00000292366</th>\n",
       "      <th>ENSG00000292372</th>\n",
       "      <th>ENSG00000292373</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>s37rep1_sum</th>\n",
       "      <td>1669</td>\n",
       "      <td>5097</td>\n",
       "      <td>158</td>\n",
       "      <td>875</td>\n",
       "      <td>103</td>\n",
       "      <td>255</td>\n",
       "      <td>223</td>\n",
       "      <td>190</td>\n",
       "      <td>469</td>\n",
       "      <td>547</td>\n",
       "      <td>...</td>\n",
       "      <td>124</td>\n",
       "      <td>560</td>\n",
       "      <td>292</td>\n",
       "      <td>9385</td>\n",
       "      <td>1</td>\n",
       "      <td>178</td>\n",
       "      <td>30</td>\n",
       "      <td>568</td>\n",
       "      <td>486</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>s37rep2_sum</th>\n",
       "      <td>1746</td>\n",
       "      <td>5407</td>\n",
       "      <td>173</td>\n",
       "      <td>850</td>\n",
       "      <td>74</td>\n",
       "      <td>269</td>\n",
       "      <td>312</td>\n",
       "      <td>167</td>\n",
       "      <td>318</td>\n",
       "      <td>586</td>\n",
       "      <td>...</td>\n",
       "      <td>124</td>\n",
       "      <td>549</td>\n",
       "      <td>315</td>\n",
       "      <td>9491</td>\n",
       "      <td>11</td>\n",
       "      <td>205</td>\n",
       "      <td>23</td>\n",
       "      <td>459</td>\n",
       "      <td>555</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>s37rep3_sum</th>\n",
       "      <td>1154</td>\n",
       "      <td>3552</td>\n",
       "      <td>136</td>\n",
       "      <td>650</td>\n",
       "      <td>27</td>\n",
       "      <td>277</td>\n",
       "      <td>166</td>\n",
       "      <td>136</td>\n",
       "      <td>475</td>\n",
       "      <td>468</td>\n",
       "      <td>...</td>\n",
       "      <td>119</td>\n",
       "      <td>683</td>\n",
       "      <td>193</td>\n",
       "      <td>7060</td>\n",
       "      <td>6</td>\n",
       "      <td>132</td>\n",
       "      <td>36</td>\n",
       "      <td>279</td>\n",
       "      <td>515</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>s32rep1_sum</th>\n",
       "      <td>1319</td>\n",
       "      <td>4640</td>\n",
       "      <td>113</td>\n",
       "      <td>481</td>\n",
       "      <td>100</td>\n",
       "      <td>238</td>\n",
       "      <td>193</td>\n",
       "      <td>91</td>\n",
       "      <td>268</td>\n",
       "      <td>333</td>\n",
       "      <td>...</td>\n",
       "      <td>55</td>\n",
       "      <td>297</td>\n",
       "      <td>273</td>\n",
       "      <td>8885</td>\n",
       "      <td>0</td>\n",
       "      <td>165</td>\n",
       "      <td>37</td>\n",
       "      <td>459</td>\n",
       "      <td>528</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>s32rep2_sum</th>\n",
       "      <td>1492</td>\n",
       "      <td>5216</td>\n",
       "      <td>97</td>\n",
       "      <td>635</td>\n",
       "      <td>91</td>\n",
       "      <td>245</td>\n",
       "      <td>315</td>\n",
       "      <td>71</td>\n",
       "      <td>297</td>\n",
       "      <td>436</td>\n",
       "      <td>...</td>\n",
       "      <td>118</td>\n",
       "      <td>395</td>\n",
       "      <td>244</td>\n",
       "      <td>8652</td>\n",
       "      <td>8</td>\n",
       "      <td>290</td>\n",
       "      <td>22</td>\n",
       "      <td>412</td>\n",
       "      <td>464</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>s32rep3_sum</th>\n",
       "      <td>1265</td>\n",
       "      <td>4412</td>\n",
       "      <td>145</td>\n",
       "      <td>618</td>\n",
       "      <td>78</td>\n",
       "      <td>157</td>\n",
       "      <td>140</td>\n",
       "      <td>54</td>\n",
       "      <td>297</td>\n",
       "      <td>398</td>\n",
       "      <td>...</td>\n",
       "      <td>59</td>\n",
       "      <td>368</td>\n",
       "      <td>141</td>\n",
       "      <td>5871</td>\n",
       "      <td>3</td>\n",
       "      <td>74</td>\n",
       "      <td>36</td>\n",
       "      <td>303</td>\n",
       "      <td>305</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6 rows × 15323 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "gene_id      ENSG00000000003  ENSG00000000419  ENSG00000000457  \\\n",
       "s37rep1_sum             1669             5097              158   \n",
       "s37rep2_sum             1746             5407              173   \n",
       "s37rep3_sum             1154             3552              136   \n",
       "s32rep1_sum             1319             4640              113   \n",
       "s32rep2_sum             1492             5216               97   \n",
       "s32rep3_sum             1265             4412              145   \n",
       "\n",
       "gene_id      ENSG00000000460  ENSG00000000971  ENSG00000001036  \\\n",
       "s37rep1_sum              875              103              255   \n",
       "s37rep2_sum              850               74              269   \n",
       "s37rep3_sum              650               27              277   \n",
       "s32rep1_sum              481              100              238   \n",
       "s32rep2_sum              635               91              245   \n",
       "s32rep3_sum              618               78              157   \n",
       "\n",
       "gene_id      ENSG00000001084  ENSG00000001167  ENSG00000001460  \\\n",
       "s37rep1_sum              223              190              469   \n",
       "s37rep2_sum              312              167              318   \n",
       "s37rep3_sum              166              136              475   \n",
       "s32rep1_sum              193               91              268   \n",
       "s32rep2_sum              315               71              297   \n",
       "s32rep3_sum              140               54              297   \n",
       "\n",
       "gene_id      ENSG00000001461  ...  ENSG00000292343  ENSG00000292344  \\\n",
       "s37rep1_sum              547  ...              124              560   \n",
       "s37rep2_sum              586  ...              124              549   \n",
       "s37rep3_sum              468  ...              119              683   \n",
       "s32rep1_sum              333  ...               55              297   \n",
       "s32rep2_sum              436  ...              118              395   \n",
       "s32rep3_sum              398  ...               59              368   \n",
       "\n",
       "gene_id      ENSG00000292345  ENSG00000292348  ENSG00000292357  \\\n",
       "s37rep1_sum              292             9385                1   \n",
       "s37rep2_sum              315             9491               11   \n",
       "s37rep3_sum              193             7060                6   \n",
       "s32rep1_sum              273             8885                0   \n",
       "s32rep2_sum              244             8652                8   \n",
       "s32rep3_sum              141             5871                3   \n",
       "\n",
       "gene_id      ENSG00000292358  ENSG00000292361  ENSG00000292366  \\\n",
       "s37rep1_sum              178               30              568   \n",
       "s37rep2_sum              205               23              459   \n",
       "s37rep3_sum              132               36              279   \n",
       "s32rep1_sum              165               37              459   \n",
       "s32rep2_sum              290               22              412   \n",
       "s32rep3_sum               74               36              303   \n",
       "\n",
       "gene_id      ENSG00000292372  ENSG00000292373  \n",
       "s37rep1_sum              486                6  \n",
       "s37rep2_sum              555                0  \n",
       "s37rep3_sum              515                0  \n",
       "s32rep1_sum              528               13  \n",
       "s32rep2_sum              464               10  \n",
       "s32rep3_sum              305                0  \n",
       "\n",
       "[6 rows x 15323 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Remove the non expressed transcripts (less than 10 counts)\n",
    "full_leng = len(df)\n",
    "df = df[df.max(axis=1) >= 10]\n",
    "\n",
    "filtr_leng = len(df)\n",
    "dropped_transcripts = full_leng - filtr_leng\n",
    "\n",
    "print(f'On {full_leng}, {dropped_transcripts} transcripts do not exceed 10 counts')\n",
    "\n",
    "df = df.T\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8eb312c3",
   "metadata": {},
   "source": [
    "### Deseq2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "3d0e9906",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Condition</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>s37rep1_sum</th>\n",
       "      <td>37°</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>s37rep2_sum</th>\n",
       "      <td>37°</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>s37rep3_sum</th>\n",
       "      <td>37°</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>s32rep1_sum</th>\n",
       "      <td>32°</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>s32rep2_sum</th>\n",
       "      <td>32°</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>s32rep3_sum</th>\n",
       "      <td>32°</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Condition\n",
       "Sample               \n",
       "s37rep1_sum       37°\n",
       "s37rep2_sum       37°\n",
       "s37rep3_sum       37°\n",
       "s32rep1_sum       32°\n",
       "s32rep2_sum       32°\n",
       "s32rep3_sum       32°"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Preapre the deseq2 metadata\n",
    "metadata = pandas.DataFrame(zip(df.index, ['37°','37°','37°','32°', '32°', '32°']),\n",
    "                            columns = ['Sample', 'Condition'])\n",
    "\n",
    "metadata = metadata.set_index('Sample')\n",
    "metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "80c649e1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AnnData object with n_obs × n_vars = 6 × 15323\n",
      "    obs: 'Condition'\n",
      "    obsm: 'design_matrix'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Fitting size factors...\n",
      "... done in 0.01 seconds.\n",
      "\n",
      "Fitting dispersions...\n",
      "... done in 4.51 seconds.\n",
      "\n",
      "Fitting dispersion trend curve...\n",
      "... done in 0.71 seconds.\n",
      "\n",
      "Fitting MAP dispersions...\n",
      "... done in 4.63 seconds.\n",
      "\n",
      "Fitting LFCs...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: total: 6.86 s\n",
      "Wall time: 21 s\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "... done in 2.52 seconds.\n",
      "\n",
      "Refitting 0 outliers.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "# define the data\n",
    "dds = DeseqDataSet(counts = df,\n",
    "                   metadata = metadata,\n",
    "                   design_factors = \"Condition\")\n",
    "print(dds)\n",
    "\n",
    "#run deseq2\n",
    "dds.deseq2()\n",
    "\n",
    "#show results, The name provided in the second element is the level that is used as baseline. \n",
    "stat_res = DeseqStats(dds, contrast = ('Condition', '32°', '37°'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "edf0495f",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Running Wald tests...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Log2 fold change & Wald test p-value: Condition 32° vs 37°\n",
      "                    baseMean  log2FoldChange     lfcSE      stat    pvalue  \\\n",
      "gene_id                                                                      \n",
      "ENSG00000000003  1417.578539        0.236871  0.190655  1.242407  0.214086   \n",
      "ENSG00000000419  4685.997108        0.422995  0.183466  2.305574  0.021134   \n",
      "ENSG00000000457   135.788628        0.025567  0.324710  0.078738  0.937241   \n",
      "ENSG00000000460   669.515399       -0.044322  0.230353 -0.192407  0.847423   \n",
      "ENSG00000000971    78.489575        0.838610  0.413537  2.027894  0.042571   \n",
      "...                      ...             ...       ...       ...       ...   \n",
      "ENSG00000292358   168.216188        0.370319  0.374895  0.987792  0.323255   \n",
      "ENSG00000292361    31.402552        0.477819  0.528571  0.903982  0.366005   \n",
      "ENSG00000292366   402.038279        0.248861  0.244840  1.016422  0.309428   \n",
      "ENSG00000292372   466.011302        0.087336  0.236827  0.368775  0.712295   \n",
      "ENSG00000292373     4.779277        2.337371  1.721920  1.357421  0.174647   \n",
      "\n",
      "                     padj  \n",
      "gene_id                    \n",
      "ENSG00000000003  0.729350  \n",
      "ENSG00000000419  0.357952  \n",
      "ENSG00000000457  0.987078  \n",
      "ENSG00000000460  0.966125  \n",
      "ENSG00000000971  0.447272  \n",
      "...                   ...  \n",
      "ENSG00000292358  0.806471  \n",
      "ENSG00000292361  0.827400  \n",
      "ENSG00000292366  0.797346  \n",
      "ENSG00000292372  0.935807  \n",
      "ENSG00000292373       NaN  \n",
      "\n",
      "[15323 rows x 6 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "... done in 1.34 seconds.\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>baseMean</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>lfcSE</th>\n",
       "      <th>stat</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000000003</th>\n",
       "      <td>1417.578539</td>\n",
       "      <td>0.236871</td>\n",
       "      <td>0.190655</td>\n",
       "      <td>1.242407</td>\n",
       "      <td>0.214086</td>\n",
       "      <td>0.729350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000419</th>\n",
       "      <td>4685.997108</td>\n",
       "      <td>0.422995</td>\n",
       "      <td>0.183466</td>\n",
       "      <td>2.305574</td>\n",
       "      <td>0.021134</td>\n",
       "      <td>0.357952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000457</th>\n",
       "      <td>135.788628</td>\n",
       "      <td>0.025567</td>\n",
       "      <td>0.324710</td>\n",
       "      <td>0.078738</td>\n",
       "      <td>0.937241</td>\n",
       "      <td>0.987078</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000460</th>\n",
       "      <td>669.515399</td>\n",
       "      <td>-0.044322</td>\n",
       "      <td>0.230353</td>\n",
       "      <td>-0.192407</td>\n",
       "      <td>0.847423</td>\n",
       "      <td>0.966125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000971</th>\n",
       "      <td>78.489575</td>\n",
       "      <td>0.838610</td>\n",
       "      <td>0.413537</td>\n",
       "      <td>2.027894</td>\n",
       "      <td>0.042571</td>\n",
       "      <td>0.447272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000292358</th>\n",
       "      <td>168.216188</td>\n",
       "      <td>0.370319</td>\n",
       "      <td>0.374895</td>\n",
       "      <td>0.987792</td>\n",
       "      <td>0.323255</td>\n",
       "      <td>0.806471</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000292361</th>\n",
       "      <td>31.402552</td>\n",
       "      <td>0.477819</td>\n",
       "      <td>0.528571</td>\n",
       "      <td>0.903982</td>\n",
       "      <td>0.366005</td>\n",
       "      <td>0.827400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000292366</th>\n",
       "      <td>402.038279</td>\n",
       "      <td>0.248861</td>\n",
       "      <td>0.244840</td>\n",
       "      <td>1.016422</td>\n",
       "      <td>0.309428</td>\n",
       "      <td>0.797346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000292372</th>\n",
       "      <td>466.011302</td>\n",
       "      <td>0.087336</td>\n",
       "      <td>0.236827</td>\n",
       "      <td>0.368775</td>\n",
       "      <td>0.712295</td>\n",
       "      <td>0.935807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000292373</th>\n",
       "      <td>4.779277</td>\n",
       "      <td>2.337371</td>\n",
       "      <td>1.721920</td>\n",
       "      <td>1.357421</td>\n",
       "      <td>0.174647</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15323 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    baseMean  log2FoldChange     lfcSE      stat    pvalue  \\\n",
       "gene_id                                                                      \n",
       "ENSG00000000003  1417.578539        0.236871  0.190655  1.242407  0.214086   \n",
       "ENSG00000000419  4685.997108        0.422995  0.183466  2.305574  0.021134   \n",
       "ENSG00000000457   135.788628        0.025567  0.324710  0.078738  0.937241   \n",
       "ENSG00000000460   669.515399       -0.044322  0.230353 -0.192407  0.847423   \n",
       "ENSG00000000971    78.489575        0.838610  0.413537  2.027894  0.042571   \n",
       "...                      ...             ...       ...       ...       ...   \n",
       "ENSG00000292358   168.216188        0.370319  0.374895  0.987792  0.323255   \n",
       "ENSG00000292361    31.402552        0.477819  0.528571  0.903982  0.366005   \n",
       "ENSG00000292366   402.038279        0.248861  0.244840  1.016422  0.309428   \n",
       "ENSG00000292372   466.011302        0.087336  0.236827  0.368775  0.712295   \n",
       "ENSG00000292373     4.779277        2.337371  1.721920  1.357421  0.174647   \n",
       "\n",
       "                     padj  \n",
       "gene_id                    \n",
       "ENSG00000000003  0.729350  \n",
       "ENSG00000000419  0.357952  \n",
       "ENSG00000000457  0.987078  \n",
       "ENSG00000000460  0.966125  \n",
       "ENSG00000000971  0.447272  \n",
       "...                   ...  \n",
       "ENSG00000292358  0.806471  \n",
       "ENSG00000292361  0.827400  \n",
       "ENSG00000292366  0.797346  \n",
       "ENSG00000292372  0.935807  \n",
       "ENSG00000292373       NaN  \n",
       "\n",
       "[15323 rows x 6 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# save results in a df\n",
    "stat_res.summary()\n",
    "\n",
    "res = stat_res.results_df\n",
    "res"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "c4a2aeab",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(15323, 9)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>Gene type</th>\n",
       "      <th>Gene description</th>\n",
       "      <th>baseMean</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>lfcSE</th>\n",
       "      <th>stat</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000000003</th>\n",
       "      <td>TSPAN6</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>tetraspanin 6 [Source:HGNC Symbol;Acc:HGNC:11858]</td>\n",
       "      <td>1417.578539</td>\n",
       "      <td>0.236871</td>\n",
       "      <td>0.190655</td>\n",
       "      <td>1.242407</td>\n",
       "      <td>0.214086</td>\n",
       "      <td>0.729350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000419</th>\n",
       "      <td>DPM1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>dolichyl-phosphate mannosyltransferase subunit...</td>\n",
       "      <td>4685.997108</td>\n",
       "      <td>0.422995</td>\n",
       "      <td>0.183466</td>\n",
       "      <td>2.305574</td>\n",
       "      <td>0.021134</td>\n",
       "      <td>0.357952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000457</th>\n",
       "      <td>SCYL3</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>SCY1 like pseudokinase 3 [Source:HGNC Symbol;A...</td>\n",
       "      <td>135.788628</td>\n",
       "      <td>0.025567</td>\n",
       "      <td>0.324710</td>\n",
       "      <td>0.078738</td>\n",
       "      <td>0.937241</td>\n",
       "      <td>0.987078</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                gene_name       Gene type  \\\n",
       "gene_id                                     \n",
       "ENSG00000000003    TSPAN6  protein_coding   \n",
       "ENSG00000000419      DPM1  protein_coding   \n",
       "ENSG00000000457     SCYL3  protein_coding   \n",
       "\n",
       "                                                  Gene description  \\\n",
       "gene_id                                                              \n",
       "ENSG00000000003  tetraspanin 6 [Source:HGNC Symbol;Acc:HGNC:11858]   \n",
       "ENSG00000000419  dolichyl-phosphate mannosyltransferase subunit...   \n",
       "ENSG00000000457  SCY1 like pseudokinase 3 [Source:HGNC Symbol;A...   \n",
       "\n",
       "                    baseMean  log2FoldChange     lfcSE      stat    pvalue  \\\n",
       "gene_id                                                                      \n",
       "ENSG00000000003  1417.578539        0.236871  0.190655  1.242407  0.214086   \n",
       "ENSG00000000419  4685.997108        0.422995  0.183466  2.305574  0.021134   \n",
       "ENSG00000000457   135.788628        0.025567  0.324710  0.078738  0.937241   \n",
       "\n",
       "                     padj  \n",
       "gene_id                    \n",
       "ENSG00000000003  0.729350  \n",
       "ENSG00000000419  0.357952  \n",
       "ENSG00000000457  0.987078  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# create a df of the result + annotation\n",
    "df_anno  = pandas.merge(annotation, res, left_index=True, right_index=True, how = 'right')\n",
    "print(df_anno .shape)\n",
    "df_anno.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "5eded6e9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# save to csv\n",
    "df_anno.to_csv(file_directory+'1) input/RNAseq/from_output/DEGs_HEK293_temp_annotated.csv', sep=',')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f4f0c1ac",
   "metadata": {},
   "source": [
    "### Significant ones"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "065c47a8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# keep padj < 0.05\n",
    "df_signif = df_anno[df_anno['padj'] < 0.05]\n",
    "print(df_signif.shape)\n",
    "df_signif[['log2FoldChange','padj','gene_name']].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4ec5bf9a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# save to .txt the genes\n",
    "# all \n",
    "numpy.savetxt(file_directory+\"3) output/allsignif_genes_HEK293.txt\", \n",
    "              df_signif['gene_name'].unique(), delimiter=\"\\t\", fmt=\"% s\")\n",
    "# positive\n",
    "numpy.savetxt(file_directory+\"3) output/top15_genes_HEK293.txt\", \n",
    "              top_15['Symbol'].unique(), delimiter=\"\\t\", fmt=\"% s\")\n",
    "\n",
    "## negative\n",
    "numpy.savetxt(file_directory+\"3) output/top5_negative_genes_HEK293.txt\", \n",
    "              topneg_5['Symbol'].unique(), delimiter=\"\\t\", fmt=\"% s\")\n",
    "\n",
    "print(top_15[['log2FoldChange','padj','gene_name']].head(), '\\n','\\n',  topneg_5[['log2FoldChange','padj','gene_name']])"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
