{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "3365a0bb",
   "metadata": {},
   "source": [
    "# Import, options"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "bcb07d2c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy, seaborn, pandas\n",
    "import matplotlib, matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "e755a2dd",
   "metadata": {},
   "outputs": [],
   "source": [
    "matplotlib.rcParams.update({'svg.fonttype' : 'none'})\n",
    "#'font.size':20, 'xtick.labelsize':20, 'ytick.labelsize':20,'axes.grid' : True, 'grid.alpha': 0.5, 'grid.linestyle' : ':',                            'figure.figsize':(8, 5), "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "979f4ea4",
   "metadata": {},
   "outputs": [],
   "source": [
    "file_directory = \"/Users/kja11/OneDrive - Menntask√Ω/PostDoc_Hypothermia/in_silico/Python/\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ee2803eb",
   "metadata": {},
   "source": [
    "# Data Download"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "67b55b58",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(15323, 10) \n",
      " Index(['gene_id', 'gene_name', 'Gene type', 'Gene description', 'baseMean',\n",
      "       'log2FoldChange', 'lfcSE', 'stat', 'pvalue', 'padj'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>Gene type</th>\n",
       "      <th>Gene description</th>\n",
       "      <th>baseMean</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>lfcSE</th>\n",
       "      <th>stat</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000100650</td>\n",
       "      <td>SRSF5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>serine and arginine rich splicing factor 5 [So...</td>\n",
       "      <td>9552.911544</td>\n",
       "      <td>1.853514</td>\n",
       "      <td>0.166420</td>\n",
       "      <td>11.137596</td>\n",
       "      <td>8.230000e-29</td>\n",
       "      <td>1.020000e-24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000181544</td>\n",
       "      <td>FANCB</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>FA complementation group B [Source:HGNC Symbol...</td>\n",
       "      <td>1798.435913</td>\n",
       "      <td>1.669421</td>\n",
       "      <td>0.202066</td>\n",
       "      <td>8.261780</td>\n",
       "      <td>1.440000e-16</td>\n",
       "      <td>8.860000e-13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000276256</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>493.431221</td>\n",
       "      <td>-1.510055</td>\n",
       "      <td>0.213682</td>\n",
       "      <td>-7.066848</td>\n",
       "      <td>1.580000e-12</td>\n",
       "      <td>6.520000e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000102317</td>\n",
       "      <td>RBM3</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>RNA binding motif protein 3 [Source:HGNC Symbo...</td>\n",
       "      <td>7155.661065</td>\n",
       "      <td>1.090947</td>\n",
       "      <td>0.175384</td>\n",
       "      <td>6.220320</td>\n",
       "      <td>4.960000e-10</td>\n",
       "      <td>1.060000e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000197245</td>\n",
       "      <td>FAM110D</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>family with sequence similarity 110 member D [...</td>\n",
       "      <td>2186.406656</td>\n",
       "      <td>1.130433</td>\n",
       "      <td>0.181118</td>\n",
       "      <td>6.241401</td>\n",
       "      <td>4.340000e-10</td>\n",
       "      <td>1.060000e-06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           gene_id gene_name       Gene type  \\\n",
       "0  ENSG00000100650     SRSF5  protein_coding   \n",
       "1  ENSG00000181544     FANCB  protein_coding   \n",
       "2  ENSG00000276256       NaN  protein_coding   \n",
       "3  ENSG00000102317      RBM3  protein_coding   \n",
       "4  ENSG00000197245   FAM110D  protein_coding   \n",
       "\n",
       "                                    Gene description     baseMean  \\\n",
       "0  serine and arginine rich splicing factor 5 [So...  9552.911544   \n",
       "1  FA complementation group B [Source:HGNC Symbol...  1798.435913   \n",
       "2                                                NaN   493.431221   \n",
       "3  RNA binding motif protein 3 [Source:HGNC Symbo...  7155.661065   \n",
       "4  family with sequence similarity 110 member D [...  2186.406656   \n",
       "\n",
       "   log2FoldChange     lfcSE       stat        pvalue          padj  \n",
       "0        1.853514  0.166420  11.137596  8.230000e-29  1.020000e-24  \n",
       "1        1.669421  0.202066   8.261780  1.440000e-16  8.860000e-13  \n",
       "2       -1.510055  0.213682  -7.066848  1.580000e-12  6.520000e-09  \n",
       "3        1.090947  0.175384   6.220320  4.960000e-10  1.060000e-06  \n",
       "4        1.130433  0.181118   6.241401  4.340000e-10  1.060000e-06  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#data DEseq HEK293\n",
    "df_RNAseq = pandas.read_csv(file_directory+'1) input/RNAseq/from_output/DEGs_HEK293_temp_annotated.csv', \n",
    "                     sep=',')\n",
    "print(df_RNAseq.shape, '\\n', df_RNAseq.columns)\n",
    "df_RNAseq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "1e8798c0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "61\n",
      "495\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>effect</th>\n",
       "      <th>Score</th>\n",
       "      <th>FDR</th>\n",
       "      <th>GeCKO_number</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PDS5B</td>\n",
       "      <td>activators</td>\n",
       "      <td>4.6</td>\n",
       "      <td>0.18</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SYT1</td>\n",
       "      <td>activators</td>\n",
       "      <td>4.5</td>\n",
       "      <td>0.18</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DDX18</td>\n",
       "      <td>activators</td>\n",
       "      <td>4.5</td>\n",
       "      <td>0.18</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>POU4F3</td>\n",
       "      <td>activators</td>\n",
       "      <td>4.4</td>\n",
       "      <td>0.18</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ZNF71</td>\n",
       "      <td>activators</td>\n",
       "      <td>4.4</td>\n",
       "      <td>0.18</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>490</th>\n",
       "      <td>CNTNAP2</td>\n",
       "      <td>repressors</td>\n",
       "      <td>2.6</td>\n",
       "      <td>0.25</td>\n",
       "      <td>523</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>491</th>\n",
       "      <td>hsa-mir-5693</td>\n",
       "      <td>repressors</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.25</td>\n",
       "      <td>525</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>492</th>\n",
       "      <td>hsa-mir-300</td>\n",
       "      <td>repressors</td>\n",
       "      <td>2.6</td>\n",
       "      <td>0.25</td>\n",
       "      <td>527</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>493</th>\n",
       "      <td>ZNF816-ZNF321P</td>\n",
       "      <td>repressors</td>\n",
       "      <td>2.6</td>\n",
       "      <td>0.23</td>\n",
       "      <td>531</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>494</th>\n",
       "      <td>hsa-mir-3129</td>\n",
       "      <td>repressors</td>\n",
       "      <td>2.6</td>\n",
       "      <td>0.25</td>\n",
       "      <td>552</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>556 rows √ó 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          gene_name      effect  Score   FDR  GeCKO_number\n",
       "0             PDS5B  activators    4.6  0.18             4\n",
       "1              SYT1  activators    4.5  0.18             5\n",
       "2             DDX18  activators    4.5  0.18             7\n",
       "3            POU4F3  activators    4.4  0.18             8\n",
       "4             ZNF71  activators    4.4  0.18             9\n",
       "..              ...         ...    ...   ...           ...\n",
       "490         CNTNAP2  repressors    2.6  0.25           523\n",
       "491    hsa-mir-5693  repressors    0.3  0.25           525\n",
       "492     hsa-mir-300  repressors    2.6  0.25           527\n",
       "493  ZNF816-ZNF321P  repressors    2.6  0.23           531\n",
       "494    hsa-mir-3129  repressors    2.6  0.25           552\n",
       "\n",
       "[556 rows x 5 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files = 'GeCKO_SP1_activators', 'GeCKO_SP1_repressors'\n",
    "\n",
    "df_GeCKO = pandas.DataFrame()\n",
    "df_list = []\n",
    "\n",
    "for file in files:\n",
    "    df_loop = pandas.read_csv(file_directory+f'1) input/GeCKO/{file}'+'.csv', sep=',')\n",
    "    df_loop = df_loop.assign(gene_name = df_loop['Gene Symbol'],\n",
    "                             effect = file.split('_')[2])\n",
    "    df_loop = df_loop[['gene_name','effect','Score','FDR','GeCKO_number']]\n",
    "    print(len(df_loop))\n",
    "    \n",
    "    df_list.append(df_loop)\n",
    "\n",
    "df_GeCKO = pandas.concat(df_list, axis=0)\n",
    "df_GeCKO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "893ada2c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "15323"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df_RNAseq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "498c6c30",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>effect</th>\n",
       "      <th>Score</th>\n",
       "      <th>FDR</th>\n",
       "      <th>GeCKO_number</th>\n",
       "      <th>RNAseq_log2fc</th>\n",
       "      <th>RNAseq_adj_pval</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PDS5B</td>\n",
       "      <td>activators</td>\n",
       "      <td>4.6</td>\n",
       "      <td>0.18</td>\n",
       "      <td>4</td>\n",
       "      <td>-0.393579</td>\n",
       "      <td>0.713558</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SYT1</td>\n",
       "      <td>activators</td>\n",
       "      <td>4.5</td>\n",
       "      <td>0.18</td>\n",
       "      <td>5</td>\n",
       "      <td>0.257288</td>\n",
       "      <td>0.863474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DDX18</td>\n",
       "      <td>activators</td>\n",
       "      <td>4.5</td>\n",
       "      <td>0.18</td>\n",
       "      <td>7</td>\n",
       "      <td>0.029512</td>\n",
       "      <td>0.983849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>POU4F3</td>\n",
       "      <td>activators</td>\n",
       "      <td>4.4</td>\n",
       "      <td>0.18</td>\n",
       "      <td>8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ZNF71</td>\n",
       "      <td>activators</td>\n",
       "      <td>4.4</td>\n",
       "      <td>0.18</td>\n",
       "      <td>9</td>\n",
       "      <td>0.537995</td>\n",
       "      <td>0.668698</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>552</th>\n",
       "      <td>CNTNAP2</td>\n",
       "      <td>repressors</td>\n",
       "      <td>2.6</td>\n",
       "      <td>0.25</td>\n",
       "      <td>523</td>\n",
       "      <td>-0.053100</td>\n",
       "      <td>0.971962</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>553</th>\n",
       "      <td>hsa-mir-5693</td>\n",
       "      <td>repressors</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.25</td>\n",
       "      <td>525</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>554</th>\n",
       "      <td>hsa-mir-300</td>\n",
       "      <td>repressors</td>\n",
       "      <td>2.6</td>\n",
       "      <td>0.25</td>\n",
       "      <td>527</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>555</th>\n",
       "      <td>ZNF816-ZNF321P</td>\n",
       "      <td>repressors</td>\n",
       "      <td>2.6</td>\n",
       "      <td>0.23</td>\n",
       "      <td>531</td>\n",
       "      <td>0.550573</td>\n",
       "      <td>0.709684</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>556</th>\n",
       "      <td>hsa-mir-3129</td>\n",
       "      <td>repressors</td>\n",
       "      <td>2.6</td>\n",
       "      <td>0.25</td>\n",
       "      <td>552</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>557 rows √ó 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          gene_name      effect  Score   FDR  GeCKO_number  RNAseq_log2fc  \\\n",
       "0             PDS5B  activators    4.6  0.18             4      -0.393579   \n",
       "1              SYT1  activators    4.5  0.18             5       0.257288   \n",
       "2             DDX18  activators    4.5  0.18             7       0.029512   \n",
       "3            POU4F3  activators    4.4  0.18             8            NaN   \n",
       "4             ZNF71  activators    4.4  0.18             9       0.537995   \n",
       "..              ...         ...    ...   ...           ...            ...   \n",
       "552         CNTNAP2  repressors    2.6  0.25           523      -0.053100   \n",
       "553    hsa-mir-5693  repressors    0.3  0.25           525            NaN   \n",
       "554     hsa-mir-300  repressors    2.6  0.25           527            NaN   \n",
       "555  ZNF816-ZNF321P  repressors    2.6  0.23           531       0.550573   \n",
       "556    hsa-mir-3129  repressors    2.6  0.25           552            NaN   \n",
       "\n",
       "     RNAseq_adj_pval  \n",
       "0           0.713558  \n",
       "1           0.863474  \n",
       "2           0.983849  \n",
       "3                NaN  \n",
       "4           0.668698  \n",
       "..               ...  \n",
       "552         0.971962  \n",
       "553              NaN  \n",
       "554              NaN  \n",
       "555         0.709684  \n",
       "556              NaN  \n",
       "\n",
       "[557 rows x 7 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_RNAseq = df_RNAseq.rename(columns = {'log2FoldChange' : 'RNAseq_log2fc',\n",
    "                                        'padj' : 'RNAseq_adj_pval'})\n",
    "\n",
    "dfmerge = pandas.merge(df_GeCKO, df_RNAseq[['gene_name','RNAseq_log2fc', 'RNAseq_adj_pval']], \n",
    "                       on= 'gene_name', how='left')\n",
    "\n",
    "dfmerge"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
